gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/SubscriberApi.java

```diff
@@ -0,0 +1,663 @@
+
+
+// AUTO-GENERATED DOCUMENTATION AND SERVICE - see instructions at the top of the file for editing.
+/**
+ * The service that an application uses to manipulate subscriptions and to
+ * consume messages from a subscription via the Pull method.
+ */
+@javax.annotation.Generated("by the veneer generator")
+public class SubscriberApi implements AutoCloseable {
+
+  // =========
+  // Constants
+  // =========
+
+  /**
+   * The default address of the service.
+   */
+  public static final String SERVICE_ADDRESS = "pubsub-experimental.googleapis.com";
+
+  /**
+   * The default port of the service.
+   */
+  public static final int DEFAULT_SERVICE_PORT = 443;
+
+
+  public static final ApiCallable<Subscription, Subscription>
+      CREATE_SUBSCRIPTION = ApiCallable.create(SubscriberGrpc.METHOD_CREATE_SUBSCRIPTION);
+  public static final ApiCallable<GetSubscriptionRequest, Subscription>
+      GET_SUBSCRIPTION = ApiCallable.create(SubscriberGrpc.METHOD_GET_SUBSCRIPTION);
+  public static final ApiCallable<ListSubscriptionsRequest, ListSubscriptionsResponse>
+      LIST_SUBSCRIPTIONS = ApiCallable.create(SubscriberGrpc.METHOD_LIST_SUBSCRIPTIONS);
+  public static final ApiCallable<DeleteSubscriptionRequest, Empty>
+      DELETE_SUBSCRIPTION = ApiCallable.create(SubscriberGrpc.METHOD_DELETE_SUBSCRIPTION);
+  public static final ApiCallable<ModifyAckDeadlineRequest, Empty>
+      MODIFY_ACK_DEADLINE = ApiCallable.create(SubscriberGrpc.METHOD_MODIFY_ACK_DEADLINE);
+  public static final ApiCallable<AcknowledgeRequest, Empty>
+      ACKNOWLEDGE = ApiCallable.create(SubscriberGrpc.METHOD_ACKNOWLEDGE);
+  public static final ApiCallable<PullRequest, PullResponse>
+      PULL = ApiCallable.create(SubscriberGrpc.METHOD_PULL);
+  public static final ApiCallable<ModifyPushConfigRequest, Empty>
+      MODIFY_PUSH_CONFIG = ApiCallable.create(SubscriberGrpc.METHOD_MODIFY_PUSH_CONFIG);
+
+
+
+  private static PageDescriptor<ListSubscriptionsRequest, ListSubscriptionsResponse, Subscription> LIST_SUBSCRIPTIONS_PAGE_DESC =
+      new PageDescriptor<ListSubscriptionsRequest, ListSubscriptionsResponse, Subscription>() {
+        @Override public Object emptyToken() {
+          return "";
+        }
+        @Override public ListSubscriptionsRequest injectToken(
+            ListSubscriptionsRequest payload, Object token) {
+          return ListSubscriptionsRequest
+            .newBuilder(payload)
+            .setPageToken((String) token)
+            .build();
+        }
+        @Override
+        public Object extractNextToken(ListSubscriptionsResponse payload) {
+          return payload.getNextPageToken();
+        }
+        @Override
+        public Iterable<Subscription> extractResources(ListSubscriptionsResponse payload) {
+          return payload.getSubscriptionsList();
+        }
+      };
+
+
+
+
+
+  private static String ALL_SCOPES[] = {
+    "https://www.googleapis.com/auth/pubsub"
+  };
+
+  public static final PathTemplate PROJECT_PATH_TEMPLATE =
```

Comment:
After thinking about this some more, I think I favor the helper functions over the PathTemplate constants. There are two benefits:

1. The barrier to understanding is lower; the helper functions have a simple interface, whereas understanding PathTemplate is more work.
2. There is less room for mistakes, since clients don't have to hard-code the field names contained within the paths. 

I find myself wondering if we should auto-generate a helper class for each type of identifier, but I'm not sure. If so, all of the helper classes could go into a single class like PublisherIdentifiers or the like.